diff -uNr portsentry-1.1.orig/portsentry_config.h portsentry-1.1/portsentry_config.h
--- portsentry-1.1.orig/portsentry_config.h	Sat Jan 11 09:00:59 2003
+++ portsentry-1.1/portsentry_config.h	Sat Jan 11 09:02:17 2003
@@ -32,7 +32,7 @@
 #define CONFIG_FILE "/etc/portsentry/portsentry.conf"
 
 /* The location of Wietse Venema's TCP Wrapper hosts.deny file */
-#define WRAPPER_HOSTS_DENY "/etc/hosts.deny"
+#define WRAPPER_HOSTS_DENY "/etc/tcpd/hosts.deny"
 
 /* The default syslog is as daemon.notice. You can also use */
 /* any of the facilities from syslog.h to send messages to (LOCAL0, etc) */
